File: euclidean/euclidean.py
def gcd(a, b):
    if a == 0: 
        return b  
    elif  b == 0:
        return a
         
    
    quotient = a // b 
    remainder = a % b
    a = (b * quotient) + remainder
    
    return gcd(b,remainder)

def gcd_common(a,b):
    if b == 0: return a 
    
    remainder = a % b
    return gcd_common(b, remainder); 

a = 270
b = 192
res = gcd(a,b)
res_common = gcd(a,b)

print(f"res: {res}")

print(f"res common: {res_common}")

File: euclidean/euclidean.java
class euclidean { 
    // ?? Recursive approach
    public static int gcd(int a, int b) {
        if (b == 0) {
            return a;
        }
        int remainder = a % b;
        return gcd(b, remainder);
    }

    // ?? Common, simpler method
    public static int gcd_common_method(int a, int b)
    {
        if (b == 0) return a; 
        int remainder = a % b; // TODO: Check if this method currently works.
        return gcd_common_method(b, remainder);
    }

    public static void main(String[] args) {
        int a = 270;
        int b = 192;
        System.out.println("GCD of " + a + " and " + b + " is: " + gcd(a, b));
        System.out.println("GCD of " + a + " and " + b + " is: " + gcd_common_method(a, b));
    }
}